src/main/java/frc/robot/subsystems/drive/SwerveModule.java

```diff
@@ -0,0 +1,237 @@
+    slot0Configs.kD = DriveCal.DRIVING_D; 
+    drivingTalon.getConfigurator().apply(slot0Configs); 
+    
+    /*errors +=
+        SparkMaxUtils.check(
+            drivingPidTmp.setOutputRange(
+                DriveCal.DRIVING_MIN_OUTPUT, DriveCal.DRIVING_MAX_OUTPUT));
+
+    errors +=
+        SparkMaxUtils.check(
+            drivingSparkMax.setIdleMode(DriveConstants.DRIVING_MOTOR_IDLE_MODE));
+    errors +=
+        SparkMaxUtils.check(
+            drivingSparkMax.setSmartCurrentLimit(
+                DriveConstants.DRIVING_MOTOR_CURRENT_LIMIT_AMPS));*/ //TODO I cannot find a Talon equivlent for this stuff
+  }
+
+  /**
+   * Burns the current settings to sparks so they keep current settings on reboot. Should be done
+   * after all settings are set.
+   */
+  public void burnFlashSparks() {
+    Timer.delay(0.005);
+    turningSparkMax.burnFlash();
+  }
+
+  /**
+   * Returns the current state of the module.
+   *
+   * @return The current state of the module.
+   */
+  public SwerveModuleState getState() {
+    // Apply chassis angular offset to the encoder position to get the position
+    // relative to the chassis.
+    return new SwerveModuleState(
+        drivingTalon.getVelocity().getValue()*DriveConstants.DRIVING_ENCODER_POSITION_FACTOR_METERS,
+        new Rotation2d(turningEncoder.getPosition() - chassisAngularOffsetRadians));
+  }
+
+  /**
+   * Returns the current position of the module.
+   *
+   * @return The current position of the module.
+   */
+  public SwerveModulePosition getPosition() {
+    // Apply chassis angular offset to the encoder position to get the position
+    // relative to the chassis.
+    return new SwerveModulePosition(
+        drivingTalon.getPosition().getValue()*DriveConstants.DRIVING_ENCODER_POSITION_FACTOR_METERS, 
+        new Rotation2d(turningEncoder.getPosition() - chassisAngularOffsetRadians));
+  }
+
+  /**
+   * Sets the desired state for the module.
+   *
+   * @param desiredState Desired state with speed and angle.
+   */
+  public void setDesiredState(SwerveModuleState desiredState) {
+    // Apply chassis angular offset to the desired state.
+    SwerveModuleState correctedDesiredState = new SwerveModuleState();
+    correctedDesiredState.speedMetersPerSecond = desiredState.speedMetersPerSecond;
+    correctedDesiredState.angle =
+        desiredState.angle.plus(Rotation2d.fromRadians(chassisAngularOffsetRadians));
+
+    // Optimize the reference state to avoid spinning further than 90 degrees.
+    SwerveModuleState optimizedDesiredState =
+        SwerveModuleState.optimize(
+            correctedDesiredState, new Rotation2d(turningEncoder.getPosition()));
+
+    // Setting global desiredState to be optimized for the shuffleboard
+    this.desiredState = optimizedDesiredState;
+
+    // Command driving and turning SPARKS MAX towards their respective setpoints.
+    drivingTalon.setControl(
+        new DutyCycleOut(optimizedDesiredState.speedMetersPerSecond));
```

Comment:
`DutyCycleOut` doesn't seem right for velocity control?

Reply:
now using VelocityVoltage